### PR TITLE
(SIMP-4164) simp: IPA integration

### PR DESCRIPTION
**Pull request by jeefberkey**
- When the host is a member of an IPA domain, do not include the
  `simp_openldap::client` class.

SIMP-4164 #comment simp/simp updated
SIMP-4301 #close

### CODE REVIEW

File: manifests/scenario/base.pp [hunk 1]
```diff
@@ -114,7 +114,10 @@
     include '::postfix'
   }
 
-  if $ldap {
+  # Even if $ldap is true, if the host is on an IPA domain, do not include
+  # simp_openldap::client
+  # @see simp/simplib lib/facter/ipa.rb
+  if $ldap or !$facts['ipa'] {
```
Comment (lnemsick-simp) [inline manifests/scenario/base.pp]: should this be 'if $ldap and !$facts['ipa'] {'

Comment (jeefberkey) [reply manifests/scenario/base.pp]: I thought we decided to ignore the ldap catalyst?

Comment (jeefberkey) [reply manifests/scenario/base.pp]: I see my error. Should be fixed now

**Comment by jeefberkey** [comment]
Puppet 5 tests will fail for now until https://github.com/rodjek/rspec-puppet/issues/642 is resolved.

**Pull request merged**